lifting-state-up/lifting-state-up.md

```diff
@@ -0,0 +1,221 @@
+| `updatedStudent`                                    | The function `props.onUpdate` ultimately accepts one argument, an updated student object. We can pass in our new `updatedStudent` object here.                                                                                                                 |
+
+## Attaching `onAttendanceButtonClick` to the Attendance Button
+
+Now that we have an event handler, let's ensure that we're _using_ this event handler!
+
+We need to attach our function to the attendance button's `onClick` attribute:
+
+<!-- prettier-ignore-start -->
+```js
+const Student = (props) => {
+
+    const onAttendanceButtonClick = () => {
+        const updatedStudent = {
+            id: props.id,
+            nameData: props.name,
+            emailData: props.email,
+            isPresentData: !props.isPresent
+        }
+
+        // Invoke the function passed in through the prop named "onUpate"
+        // This function is referenced by the name "updateStudent" in App
+        props.onUpdate(updatedStudent);
+    }
+
+    const nameColor = props.isPresent ? 'green' : 'red';
+
+    return (
+        <div>
+            <ul>
+                <li className={nameColor}>Nickname: {props.name}</li>
+                <li>Email: {props.email}</li>
+            </ul>
+            <button onClick={onAttendanceButtonClick}>Toggle if {props.name} is present</button>
+        </div>
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+## Verifying
+
+Let's see if our app works still!
+
+Let's verify that user interactions affect the state of the app using React Dev Tools.
+
+We should first observe that the `App` component indeed keeps `studentData` on state.
+
+![](../assets/lifting-state-up_lifting-state-up_app-state.png)
+
+And clicking on the attendance buttons updates the state in `App`!
+
+![](../assets/lifting-state-up_lifting-state-up_app-state-updated.png)
+
+![]()
+
+We did it! 🎉🎉
+
+Let's observe two more things:
+
+`StudentList` indeed has two `props`: a `students` array with the student data, and `onUpdateStudent`, which is a function.
+
+![](../assets/lifting-state-up_lifting-state-up_studentlist-detail.png)
+
+Each `Student` component has five props:
+
+1. `email`
+1. `id`
+1. `isPresent`
+1. `name`
+1. `onUpdate`, whose value is a function originally named `updateStudentData`
+
+![](../assets/lifting-state-up_lifting-state-up_student-detail.png)
+
+## Did We Accomplish "Single Source of Truth"?
+
+We've accomplished the following:
+
+1. We set `App` as the "single source of truth" of student data, and removed state from `Student`
+1. We created a function that updates the state of `studentData` in `App`. Then, we passed that function down to `Student` through `props`.
+1. We created an `onClick` event handler in `Student`. When the attendance button is called, it invokes the function that updates the state of `studentData` in `App`.
```

Comment:
existing phrasing sounds like the function we wrote was called onClick
we've already specified the function, so using "that" sounds like a determinant, as though it had multiple functions to pick from.
```suggestion
1. We created an event handler for the `onClick` event in `Student`. When the attendance button is clicked, it invokes the function, which updates the state of `studentData` in `App`.
```